### PR TITLE
Fix Leak: TRY_CATCH_ERROR must release the error manually because thi…

### PR DESCRIPTION
**Pull request by stefanpenner**
…ngs...

### CODE REVIEW

File: lib/rsvp/-internal.js [hunk 1]
```diff
@@ -204,6 +204,7 @@ export function invokeCallback(settled, promise, callback, detail) {
     if (value === TRY_CATCH_ERROR) {
       failed = true;
       error = value.error;
```
Comment (runspired) [inline lib/rsvp/-internal.js]: is this still meaningful with the below?

Comment (stefanpenner) [inline lib/rsvp/-internal.js]: The `value = null` looks like my original but failed trying to release, good catch.

Comment (runspired) [inline lib/rsvp/-internal.js]: now that you've removed `value = null` this makes much more sense :D disregard the comment.

**Comment by stefanpenner** [comment]
test failure is unrelated node 4 test failure due to a transitive dependency...

**Comment by stefanpenner** [comment]
released as 3.4.0

**Comment by cowboyd** [comment]
Is there any way to monkeypatch this on an application running Ember 2.8 (which bundles 3.2.1) ?

**Comment by stefanpenner** [comment]
@cowboyd if you are loading ember via bower still, you could fork that version of ember and point your bower to it.

Or are you asking if we can/should backport?

**Pull request merged**